text/0000-c_locale_ctype.md

```diff
@@ -0,0 +1,153 @@
+ `0x21`..`0x2F`  | 0         | 0         | 0         | 0         | 0         | 1         | 0         | 1         | 1         | 0         | 0         | 0
+ `0x30`..`0x39`  | 1         | 0         | 0         | 0         | 1         | 1         | 0         | 1         | 0         | 0         | 0         | 1
+ `0x3a`..`0x40`  | 0         | 0         | 0         | 0         | 0         | 1         | 0         | 1         | 1         | 0         | 0         | 0
+ `0x41`..`0x46`  | 1         | 1         | 0         | 0         | 0         | 1         | 0         | 1         | 0         | 0         | 1         | 1
+ `0x47`..`0x5A`  | 1         | 1         | 0         | 0         | 0         | 1         | 0         | 1         | 0         | 0         | 1         | 0
+ `0x5B`..`0x60`  | 0         | 0         | 0         | 0         | 0         | 1         | 0         | 1         | 1         | 0         | 0         | 0
+ `0x61`..`0x66`  | 1         | 1         | 0         | 0         | 0         | 1         | 1         | 1         | 0         | 0         | 0         | 1
+ `0x67`..`0x7A`  | 1         | 1         | 0         | 0         | 0         | 1         | 1         | 1         | 0         | 0         | 0         | 0
+ `0x7B`..`0x7E`  | 0         | 0         | 0         | 0         | 0         | 1         | 0         | 1         | 1         | 0         | 0         | 0
+ `0x7F`          | 0         | 0         | 0         | 1         | 0         | 0         | 0         | 0         | 0         | 0         | 0         | 0
+
+## Character Case Mapping Function Set
+[Character Case Mapping Function Set]: #character-case-mapping-function-set
+
+ Character Range | `tolower` | `toupper`
+:---------------:|:---------:|:---------:
+ `0x00`..`0x40`  | 0         | 0
+ `0x41`..`0x5A`  | 1         | 0
+ `0x5B`..`0x60`  | 0         | 0
+ `0x61`..`0x7A`  | 0         | -1
+ `0x7B`..`0x7F`  | 0         | 0
+
+## Features
+[Features]: #features
+
+This RFC also introduces two features to be used as configuration options:
+
+### `ctype`
+[`ctype`]: #ctype
+
+This feature brings `<ctype.h>` functions into scope. This feature however
+does not specify behaviour of the functions.
+
+### `c_locale_ctype`
+[`c_locale_ctype`]: #c_locale_ctype
+
+This feature allows running `<ctype.h>` with `"C"` locale, as defined above.
+This feature depends on the feature [`ctype`].
+
+# Drawbacks
+[Drawbacks]: #drawbacks
+
+None
+
+# Rationale and alternatives
+[Rationale and alternatives]: #rationale-and-alternatives
+
+None
+
+# Unresolved questions
+[Unresolved questions]: #unresolved-questions
+
+* [ ] Should we mark the functions as `#[unstable]` until a future RFC
+introduces the `locale.h`?
+* [ ] How do we deal with the undefined behaviour for invalid values.
+* [ ] Should this RFC mandate checking for `EOF` right now for the functions.
```

Comment:
My thoughts is to not consider it now, instead leaving a `// TODO` comment, which we resolve when I/O is introduced